File: src/TheEight.WebApp/client/app/water-events/boat-lineup-planner/records/Boat.js
import { Record, Map } from "Immutable";

const defaults = {
    boatId: "",
    title: "",
    isCoxed: false,
    seatCount: 0,
    seatAssignments: Map()
};

export default class extends Record(defaults) {
    unassignSeat(position) {
        return this.set("seatAssignments", this.seatAssignments.delete(position));
    }

    assignAttendee(position, attendeeId) {
        return this.set("seatAssignments", this.seatAssignments.set(position, attendeeId));
    }

    isAttendeeInBoat(attendeeId) {
        return this.seatAssignments.contains(attendeeId);
    }

    isSeatAssigned(position) {
        return this.seatAssignments.has(position);
    }
}

File: src/TheEight.Domain/WaterEvents/Boat.cs
using System;

namespace TheEight.Domain.WaterEvents
{
    public class Boat
    {
        public Guid BoatId { get; set; }
        public string Title { get; set; }
        public int SeatCount { get; set; }
        public bool IsCoxed { get; set; }
    }
}